media-libs/dav1d/dav1d-0.5.2.recipe

```diff
@@ -0,0 +1,77 @@
+SUMMARY="A new AV1 cross-platform decoder"
+DESCRIPTION="dav1d is a new AV1 cross-platform decoder, open-source, \
+and focused on speed and correctness. It supports all features from AV1, \
+including all subsampling and bit-depth parameters."
+HOMEPAGE="https://code.videolan.org/videolan/dav1d"
+COPYRIGHT="2018-2019, VideoLAN and dav1d authors"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://code.videolan.org/videolan/dav1d/-/archive/$portVersion/dav1d-$portVersion.tar.bz2"
+CHECKSUM_SHA256="1780199d71e62a82ca1bfca2d71da5c599c7e1edc68c72f6fc3c0c25ce1880b7"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64 ?arm ?ppc"
+SECONDARY_ARCHITECTURES="x86"
+
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+libVersion="3.1.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	dav1d$secondaryArchSuffix = $portVersion
+	cmd:dav1d$commandSuffix
+	lib:libdav1d$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	dav1d${secondaryArchSuffix}_devel = $portVersion
+	devel:libdav1d$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	dav1d$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:meson >= 0.47
+	cmd:nasm >= 2.13.02
+	cmd:ninja
+	"
+
+BUILD()
+{
+	rm -f -rf build
+	mkdir build && cd build
+
+	meson .. \
```

Comment:
Again, can you switch to `meson build`? It'd remove the need for creating directories.

Reply:
I don't think so.
When I use meson build, it works perfectly the first time the source code is extracted and built.
However, it gives error messages about not being able to find ninja when running haikuporter the next time.

Reply:
Run it, package it (failure or success in the packaging does not matter).
Run it again, **without** removing the build directory, it gives errors.

Reply:
That's what I did.

Reply:
Oh, I'm sorry.
Running the FIRST time it fails

> Program objcopy found: YES (/bin/objcopy)
> Build targets in project: 15
> Found ninja-1.9.0 at /bin/ninja
> ninja: error: loading 'build.ninja': No such file or directory
> Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 1
> Error: Build has failed - stopping.
> ~/haikuports/media-libs/dav1d>

Reply:
Maybe it'd work if you remembered to add `-C build` to ninja?

Reply:
Works this time. Pushed everything. Thanks!